src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -169,7 +168,9 @@ trait IterableMonoTransforms[+A, +Repr] extends Any {
   def drop(n: Int): Repr = fromIterableWithSameElemType(View.Drop(coll, n))
 
   /** The rest of the collection without its first element. */
-  def tail: Repr = drop(1)
+  def tail: Repr =
+    if (coll.isEmpty) drop(1)
```

Comment:
Oops, yes. It's too hot here (37 celsius at 9am this morning!) Can't think clearly.